### PR TITLE
Implement retry flags per protocol doc

### PR DESCRIPTION
**Pull request by jcorbin**

### CODE REVIEW

File: node/request.js [hunk 1]
```diff
@@ -195,20 +204,28 @@ TChannelRequest.prototype.shouldRetry = function shouldRetry(err, res, arg2, arg
...
             case 'tchannel.canceled':
                 return false;
 
-            case 'tchannel.socket':
-            case 'tchannel.timeout':
             case 'tchannel.busy':
             case 'tchannel.declined':
-            case 'tchannel.unexpected':
-            case 'tchannel.protocol':
                 return true;
```
Comment (jcorbin) [inline node/request.js]: cc @mranney @mmihic does this seem like a missed flag? or intentionally left out?

File: node/request.js [hunk 2]
```diff
@@ -195,20 +204,28 @@ TChannelRequest.prototype.shouldRetry = function shouldRetry(err, res, arg2, arg
...
 
+            case 'tchannel.timeout':
+                return !!self.options.retryFlags.onTimeout;
+
+            case 'tchannel.socket':
+            case 'tchannel.protocol':
+            case 'tchannel.unexpected':
```
Comment (jcorbin) [inline node/request.js]: cc @mranney @mmihic it's correct to treat all 3 the same here right?

File: node/request.js [hunk 3]
```diff
@@ -195,20 +204,28 @@ TChannelRequest.prototype.shouldRetry = function shouldRetry(err, res, arg2, arg
...
     if (err) {
         switch (err.type) {
             case 'tchannel.bad-request':
             case 'tchannel.canceled':
                 return false;
 
-            case 'tchannel.socket':
-            case 'tchannel.timeout':
             case 'tchannel.busy':
             case 'tchannel.declined':
```
Comment (jcorbin) [inline node/request.js]: cc @mranney @mmihic does this seem like a missed flag? or intentionally left out?

File: node/request.js (hunk 2 above)
Comment (mmihic) [inline node/request.js]: yes

File: node/request.js (hunk 3 above)
Comment (mmihic) [inline node/request.js]: Sorry, which flag?

Comment (jcorbin) [inline node/request.js]: The busy/declined path, it's hardwired to true here.

Seems a bit of a minor miss to me:
- would be better explicate a flag, say `"d"` for declined
- change the default retryflags then to be `"cd"`

We may not feel the lack of it much, but implementing the code seemed to point it out to me as an oversight.

**Comment by Raynos** [comment]
lgtm.

**Pull request merged**